File: Design Patterns/Command Pattern/Java/com.faculdadevincit.designpatterns.commandpattern/src/LampadaDesligarCommand.java
public class LampadaDesligarCommand implements ICommand {

    private Lampada lampada;

    public LampadaDesligarCommand(Lampada lampada) {
        this.lampada = lampada;
    }

    @Override
    public void executar() {
        lampada.desligar();
    }

    @Override
    public void desfazer() {
        lampada.ligar();
    }

}

File: Design Patterns/Command Pattern/CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/LampadaDesligarCommand.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class LampadaDesligarCommand : ICommand
{
    public Lampada lampada { get; private set; }

    public LampadaDesligarCommand(Lampada lampada)
    {
        this.lampada = lampada;
    }

    public void Executar() => lampada.desligar();

    public void Desfazer() => lampada.ligar();
}
